### PR TITLE
JBPM-5716: Stunner - Shapes and controls broken features as for some mouse interactions.

### PR DESCRIPTION
**Pull request by romartin**
Hey @manstis , @hasys 
This commit includes a few fixes but that should be included for the release! 
See [the Jira ticket](https://issues.jboss.org/browse/JBPM-5716)
@hasys  sorry if this time cannot add missing tests for existing classes, will keep adding tests on further iterations..
@manstis this commit includes the fix for the canvas preview type, as commented, a partial fix yet...
Thanks!

### CODE REVIEW

**Comment by romartin** [comment]
it includes two type renamings, that's why a few many files appear to be changed...

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/shape/view/ViewEventHandlerManager.java [hunk 1]
```diff
@@ -125,6 +130,12 @@ public void addHandlersRegistration(final ViewEventType type,
         if (ViewEventType.MOUSE_DBL_CLICK.equals(type)) {
             return registerDoubleClickHandler((ViewHandler<ViewEvent>) eventHandler);
         }
+        if (ViewEventType.MOUSE_ENTER.equals(type)) {
+            return registerEnterHHandler((ViewHandler<ViewEvent>) eventHandler);
```
Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/shape/view/ViewEventHandlerManager.java]: Typo.. ```...EnterHHandler``` (double "H")

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/shape/view/ViewEventHandlerManager.java [hunk 2]
```diff
@@ -198,11 +209,45 @@ protected GestureEventImpl buildGestureEvent(final AbstractNodeGestureEvent even
                                     event.getRotation());
     }
 
+    protected HandlerRegistration[] registerEnterHHandler(final ViewHandler<ViewEvent> eventHandler) {
```
Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/shape/view/ViewEventHandlerManager.java]: Typo.. ```...EnterHHandler``` (double "H")

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/controls/actions/CanvasNameEditionControlImpl.java [hunk 3]
```diff
@@ -103,41 +103,39 @@ public void register(final Element element) {
...
-                            @Override
-                            public void handle(TextOverEvent event) {
-                                canvasHandler.getAbstractCanvas().getView().setCursor(AbstractCanvas.Cursors.TEXT);
-                            }
-                        };
-                        if (hasEventHandlers.supports(ViewEventType.TEXT_OVER) &&
-                                hasEventHandlers.supports(ViewEventType.TEXT_OUT)) {
-                            hasEventHandlers.addHandler(ViewEventType.TEXT_OVER,
+                                        clickHandler);
+                        // Change mouse cursor, if shape supports it.
```
Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/controls/actions/CanvasNameEditionControlImpl.java]: What restores the cursor when exists the text item? The handler is called ```overHandler``` which suggests the cursor is meant to change on a "mouse over" event; however the handler class is ```TextEnterHandler``` which suggests you need a ```TextExitHandler``` to restore the cursor too? Which is it; mouse over or mouse enter and mouse exit?

Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/controls/actions/CanvasNameEditionControlImpl.java]: Hey @manstis , well agree.. using `overHandler` as name for the `TextEnterHandler` probably is not the best option. So the idea is that there are 2 event handlers for the text - enter and exit. On the enter callback, the cursor is set to hand, while when existing text the cursor is back to the `auto` value. 
So is it good for just just renaming the `overHandler` and `outHandler` by `enterHandler` and `exitHandler` ? Or you see some other issues? Thanks Mike!

Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/controls/actions/CanvasNameEditionControlImpl.java]: @romartin Hi.. nope that's fine.. I didn't see the ```final TextExitHandler outHandler = new TextExitHandler()``` code in the PR diffs; but now I see it.... so all good!

Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/controls/actions/CanvasNameEditionControlImpl.java]: @manstis hhehe np! anyway as for the typos above, it worths renaming that names as well. Just rebasing and comitting again, thanks!

**Comment by romartin** [comment]
btw @manstis this fix about text makes the text edition more intuitive... as before, you had to dbl-click on the shape (not the text) in order to display the name editor popup, so I saw people trying to click on text for thousands of times with no success hehe... now, once you get over the text, the cursor changes, at that point, you can dbl-click and the name editor popup appears.... so will see how it goes... but I think it will be clear for users

**Comment by romartin** [comment]
@manstis new commit added. Thanks!

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by romartin** [comment]
@manstis tbh still not sure about what the build error is about :( it says all good, just a few warns...

**Pull request merged**